HznApplication/HznApp.cpp

```diff
@@ -2,55 +2,234 @@
 
 #include <HorizonEngine.h>
 #include <HznEntryPoint.h>
-
 #include "HznApp.h"
+#include <commdlg.h>
+#include "FileManagement/ProjectFile.h"
+#include "SceneManagement/Scene.h"
 
+std::shared_ptr<Hzn::App> Hzn::createApp()
+{
+	auto app = std::make_shared<HznApp>();
+	app->addLayer(new SampleLayer());
+	//app->addLayer(new EditorLayer());
+	return app;
+}
 
+// *********** SAMPLE LAYER **********
 
+SampleLayer::SampleLayer(const std::string& name) : Layer(name)
+{
+	std::string gridVertexShader = R"(
+		#version 420 core
+		layout (location = 0) in vec3 a_Pos;
+
+		uniform mat4 model;
+		uniform mat4 view;
+		uniform mat4 projection;
+		
+		void main()
+		{
+			gl_Position = projection * view * model * vec4(a_Pos, 1.0f);
+		}
+	)";
 
-//open & save file dependencies for win32 API
+	std::string gridFragmentShader = R"(
+		# version 420 core
+		
+		uniform vec4 a_Color;
+		out vec4 fragmentColor;
 
-#include <commdlg.h>
-#include "FileManagement/ProjectFile.h"
-#include "SceneManagement/Scene.h"
+		void main()
+		{
+			fragmentColor = a_Color;
+		}
+	)";
 
+	std::string textureVertexShader = R"(
+		#version 420 core
+		layout (location = 0) in vec3 a_Pos;
+		layout (location = 1) in vec2 a_TexCoord;
 
+		uniform mat4 model;
+		uniform mat4 view;
+		uniform mat4 projection;
 
+		out vec2 v_TexCoord;
+		
+		void main()
+		{
+			gl_Position = projection * view * model * vec4(a_Pos, 1.0f);
+			v_TexCoord = a_TexCoord;
+		}
+	)";
 
+	std::string textureFragmentShader = R"(
+		# version 420 core
+		
+		in vec2 v_TexCoord;
+		out vec4 fragmentColor;
 
+		uniform sampler2D f_Texture;
 
-std::shared_ptr<Hzn::App> Hzn::createApp()
-{
-	return std::make_shared<HznApp>();
-}
+		void main()
+		{
+			fragmentColor = texture(f_Texture, v_TexCoord);
+		}
+	)";
 
-// *********** SAMPLE LAYER **********
+	Hzn::BufferLayout layout =
+	{
+		{Hzn::ShaderDataType::Vec3f, "a_Pos"},
+		{Hzn::ShaderDataType::Vec2f, "a_TexCoord"}
+	};
+
+	std::vector<float> vertices = {
```

Comment:
What are there's parameters for?

Reply:
The layout of the vertex Buffer. In OpenGL when you feed vertex data (array of floats) to the GPU, you have to tell the GPU how your vertex data is arranged. Like in a vertex you can have attributes like (texture coordinates, color, position, etc). This defines your layout and you have to tell this to the GPU.